tests/sles4sap/hana_install.pm

```diff
@@ -0,0 +1,123 @@
+# SUSE's SLES4SAP openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Install HANA via command line. Verify installation with
+# sles4sap/hana_test
+# Maintainer: Alvaro Carvajal <acarvajal@suse.com>
+
+use base 'sles4sap';
+use strict;
+use warnings;
+use testapi;
+use version_utils 'is_sle';
+
+sub upload_install_log {
+    script_run "tar -zcvf /tmp/hana_install.log.tgz /var/tmp/hdb*";
+    upload_logs "/tmp/hana_install.log.tgz";
+}
+
+sub run {
+    my ($self) = @_;
+    my ($proto, $path) = $self->fix_path(get_required_var('MEDIA'));
+
+    my $sid      = get_required_var('INSTANCE_SID');
+    my $instid   = get_required_var('INSTANCE_ID');
+    my $password = 'Qwerty_123';
+    set_var('PASSWORD', $password);
+
+    $self->select_serial_terminal;
+    my $RAM = $self->get_total_mem();
+    die "RAM=$RAM. The SUT needs at least 24G of RAM" if $RAM < 24000;
+
+    # Add host's IP to /etc/hosts
+    my $netdevice = get_var('SUT_NETDEVICE', 'eth0');
+    assert_script_run "echo \$(ip -4 addr show dev $netdevice | sed -rne '/inet/s/[[:blank:]]*inet ([0-9\\.]*).*/\\1/p') \$(hostname) >> /etc/hosts";
+
+    # This installs HANA. Start by configuring the appropiate SAP profile
+    $self->prepare_profile('HANA');
+
+    # Copy media
+    my $tout = get_var('HANA_INSTALLATION_TIMEOUT', 2700);    # Timeout for HANA installation commands. Defaults to NW's timeout times 2
+    $self->copy_media($proto, $path, $tout, '/sapinst');
+
+    # Mount points information: use the same paths and minimum sizes as the wizard
+    my %mountpts = (
+        hanadata   => {mountpt => '/hana/data',         size => '24g'},
+        hanalog    => {mountpt => '/hana/log',          size => '12g'},
+        hanashared => {mountpt => '/hana/shared',       size => '24g'},
+        usr_sap    => {mountpt => "/usr/sap/$sid/home", size => '50g'}
+    );
+
+    # Partition disks for Hana
+    if (check_var('HANA_PARTITIONING_BY', 'yast')) {
+        my $yast_partitioner = is_sle('15+') ? 'sap_create_storage_ng' : 'sap_create_storage';
+        assert_script_run "yast $yast_partitioner /usr/share/YaST2/include/sap-installation-wizard/hana_partitioning.xml";
```

Comment:
I just saw today on some YaST tests that now (since when?) yast modules should be called using this kind of syntax (with some actions before the `wait_serial` if needed but not here):
`my $module_name = y2_module_consoletest::yast2_console_exec(yast2_module => "$yast_partitioner");
wait_serial("$module_name-0", 120) || die "'yast2 $yast_partitioner' didn't finish";
`
Don't forget to add `use y2_module_consoletest;` before ;-)

Again sorry, I just saw this today...

Reply:
Wow. That looks cumbersome....

Reply:
AFAICS `y2_module_consoletest` is a base class for `yast2` tests. The sles4sap tests have their own base class, which I wouldn't recommend changing. I'll try importing the class, but I'm not really sure we should use that here, with a YaST module that's not interactive. Let' see

Reply:
Maybe yes, I didn't tried myself.

Reply:
Tested this, and the only way I was able to make it work was with more changes than the call to the method:

1. I had to add an option to `yast2_console_exec()` in `lib/y2_module_consoletest` so the method can receive command line parameters for the module.
2. I needed to use `select_console 'root-console'` instead of `$self->select_serial_terminal` as it seems the call to `wait_serial` after the call to `yast2_console_exec()` loses part of the serial output, including the regexp with the return code.

If this change is necessary, I would propose to do it in a separate PR as the first item requires changes to modules outside of sles4sap scope and should be reviewed by their maintainers.

Reply:
For me this change is not mandatory, it would have been better to use dedicated function (like we do with `systemctl`) but only if it's for something better/easier.

As you mentioned, this YaST call is not interactive in our case, so `assert_script_run` is enough.